### PR TITLE
require metamascara

### PR DESCRIPTION
**Pull request by frankiebee**
WOOF! require mascara module

### CODE REVIEW

File: app/scripts/background.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ window.log = log
 log.setDefaultLevel(METAMASK_DEBUG ? 'debug' : 'warn')
 
 const platform = new ExtensionPlatform()
+platform.context = 'extension'
```
Comment (kumavis) [inline app/scripts/background.js]: lets put this in the platform class

Comment (kumavis) [reply app/scripts/background.js]: how about `isExtension: true` in the `ExtensionPlatform` class definition

**Review by kumavis — CHANGES_REQUESTED** [review]

File: app/scripts/metamask-controller.js [hunk 2]
```diff
@@ -196,7 +196,7 @@ module.exports = class MetamaskController extends EventEmitter {
       },
       // rpc data source
       rpcUrl: this.networkController.getCurrentRpcAddress(),
-      originHttpHeaderKey: 'X-Metamask-Origin',
```
Comment (kumavis) [inline app/scripts/metamask-controller.js]: lets re-enable setting this flag

Comment (frankiebee) [reply app/scripts/metamask-controller.js]: 👍

**Review by kumavis — APPROVED** [review]
good to go after the `x-metamask-origin` is re-enabled

**Pull request merged**